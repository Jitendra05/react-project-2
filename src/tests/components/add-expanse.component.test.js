import React from 'react';
import {shallow} from 'enzyme';
import {AddExpanseComponent} from '../../components/add-expanse.component';
import {expanses} from '../fixtures/expanses-data';

let startAddExpanseSpy, historySpy, wrapper;

beforeEach(()=>{
  startAddExpanseSpy =  jest.fn();
  historySpy = {push:jest.fn()};
  wrapper = shallow(<AddExpanseComponent startAddExpanse={startAddExpanseSpy} history={historySpy}/>);
});

test('Should render AddExpanseComponent correctly',()=>{
  expect(wrapper).toMatchSnapshot();
});

test('Should handle onSubmit',()=>{
  wrapper.find('ExpanseForm').prop('onSubmit')(expanses[1]);
  expect(historySpy.push).toHaveBeenLastCalledWith("/");
  expect(startAddExpanseSpy).toHaveBeenLastCalledWith(expanses[1]);
});