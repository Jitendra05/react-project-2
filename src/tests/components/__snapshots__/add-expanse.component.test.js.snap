// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AddExpanseComponent correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="container"
    >
      <h2
        className="page-header__title"
      >
        Add Expense
      </h2>
    </div>
  </div>
  <div
    className="container"
  >
    <ExpanseForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
